### DISCUSSION TITLE
Escaped delimiters

### DISCUSSION THREAD

**Discussion by elavy-harris** [nextgenhealthcare/connect#5180] [Q&A]
If PID.9.1 contains an escaped delimiter as part of the data (e.g., `abc\E\def`) and I reference `msg['PID']['PID.9']['PID.9.1'].toString()`, I get the literal value from the HL7 message, not the intended text with the unescaped character.

Is there a way to get that unescaping to occur automatically?  To do it manually requires collecting MSH.1 and MSH.2 and using their contents to figure out if something is escaped and what the unescaped value is.  Theoretically, this would need to be done for every piece of data.

**Comment by pacmano1** [comment]
Not sure this has changed much from https://github.com/nextgenhealthcare/connect/issues/1262
